src/frontend/public/common/blockchain/wallet.js

```
@@ -20,6 +21,28 @@ class Wallet {
         return await this.generateAccountFromMnemonicEthers(mnemonic, password, progressCallback);
     }
 
+    /**
+     * Function to recover wallet from mnemonic and public address.
+     * @param {String} mnemonic - User's mnemonic words.
+     * @param {String} password - User's password as string.
+     * @param {String} address - User's public address.
+     * @param {String} progressCallback - callback function for tracking progress
+     * @returns {Object} properties: walletAddress, walletJSON, mnemonic
+     */
+
+    static async recoverFromMnemonic(mnemonic, password, address, progressCallback) {
+        baseValidators.validatePassword(password);
+        baseValidators.validateAddress(address, ERROR.INVALID_RECOVERED_ADDRESS);
```

The message here is not correct, if this validation fails it should throw provided address is invalid